### ISSUE TITLE
installer glitch

### ISSUE THREAD

**Issue by pameyer** [IQSS/dataverse#3515]
as of 88d2e50b2f08539c4980e1222f17c208776fe286; the installer generated by `mvn package` fails with:
Command line: /usr/local/glassfish4/bin/asadmin deploy dataverse.war
remote failure: Error occurred during deployment: Exception while loading the app : Error in linking security policy for dataverse -- Inconsistent Module State. Please see server.log for more details.

PER01003: Deployment encountered SQL Exceptions:
	PER01000: Got SQLException executing statement "CREATE INDEX INDEX_DATAVERSE_fk_dataverse_id ON DATAVERSE (fk_dataverse_id)": org.postgresql.util.PSQLException: ERROR: column "fk_dataverse_id" does not exist
	PER01000: Got SQLException executing statement "CREATE INDEX INDEX_DATAFILE_md5 ON DATAFILE (md5)": org.postgresql.util.PSQLException: ERROR: column "md5" does not exist
Command deploy failed.
Failed to deploy the application! WAR file: dataverse.war.
(exit code: 256)
Aborting.

---
Guessing this is related to #3354; haven't done extensive investigation to confirm.

**Comment by pdurbin** [comment]
Hmm, I deployed fine (with warnings) as of 75a4146 (the latest in the "develop" branch as of this writing) but I see what you mean about the `ERROR: column "md5" does not exist` warning. (Indexes were added in #1880 but I think this is unrelated.)

```
Command start-domain executed successfully.

PER01003: Deployment encountered SQL Exceptions:
	PER01000: Got SQLException executing statement "CREATE INDEX INDEX_DATAVERSE_fk_dataverse_id ON DATAVERSE (fk_dataverse_id)": org.postgresql.util.PSQLException: ERROR: column "fk_dataverse_id" does not exist
	PER01000: Got SQLException executing statement "CREATE INDEX INDEX_DATAFILE_md5 ON DATAFILE (md5)": org.postgresql.util.PSQLException: ERROR: column "md5" does not exist
Command deploy completed with warnings.
```

Please see https://build.hmdc.harvard.edu:8443/job/phoenix.dataverse.org-deploy-develop/68/console for details.

Actually, hmm, `md5` is still being referenced here: https://github.com/IQSS/dataverse/blob/2ab8664c267bd4c79430e102ddc311b56c31357b/src/main/java/edu/harvard/iq/dataverse/DataFile.java#L49 . I bet that's it. This is related to #3303.

**Comment by pdurbin** [comment]
@pameyer please see #2944 which has the following todo item:

Update http://guides.dataverse.org/en/4.3.1/installation/installation-main.html#troubleshooting to mention how restarting Glassfish can help. See https://help.hmdc.harvard.edu/Ticket/Display.html?id=236419 . '"Inconsistent module state" [usually] means "keep restarting glassfish and redeploying" until it works. If it's still not working after 3 restarts, delete the entire contents of the "domain1/generated" directory, and keep restarting.'

Whatever you were doing, can you please try again and see if the war file continues to fail to deploy?

**Comment by pameyer** [comment]
In my hands, this is repeatable.  removed dvndb and re-reran the installer, same error.

This can be closed if it's not something that shows up in the upcoming release.

**Comment by pdurbin** [comment]
I just ran `vagrant up` on adbe63d and it worked fine. I can log in as "dataverseAdmin" at http://localhost:8888 per the screenshot below.

![screen shot 2016-12-09 at 12 14 44 pm](https://cloud.githubusercontent.com/assets/21006/21057836/301bb158-be09-11e6-9415-49b8604fbc19.png)

It *is* confusing that it says http://localhost:8080 in the final output rather than http://localhost:8888/ but hopefully people are following the docs at http://guides.dataverse.org/en/4.5.1/installation/prep.html#vagrant-for-testing-only which tells the correct port.

```
==> standalone: You should now have a running DVN instance at
==> standalone:   http://localhost:8080
```

I'm thinking I'll add and extra message to the bottom to explain this.

Here's the output of the adbe63d run: [output-adbe63d.txt](https://github.com/IQSS/dataverse/files/642801/output-adbe63d.txt)

**Comment by pdurbin** [comment]
Hmm, I ran `vagrant up` again and got `==> standalone: cannot stat `/etc/shibboleth/attribute-map.xml'` which means Shibboleth couldn't be installed. @pameyer mentioned this as well and it seems like those RPM repos have been flaky lately based on chatter at https://irclog.perlgeek.de/shibboleth/2016-11-23 . I don't actually use Vagrant for Shib dev anymore (despite what the Dev Guide says) so I should probably rip it out of Vagrant.

**Comment by pdurbin** [comment]
There's still a lingering warning/error about jhove but it doesn't seem to be fatal. It says `==> standalone: sh: /home/glassfish/glassfish4/glassfish/domains/domain1/config/../../conf/jhove/jhove.conf: No such file or directory`

**Comment by pdurbin** [comment]
@djbrooke I just make pull request #3520 and I'm assigning this issue to you to review the doc changes. Thanks!

**Comment by pdurbin** [comment]
Here's the output from `vagrant up` as of 54fa0ec (pull request #3520): [output-54fa0ec.txt](https://github.com/IQSS/dataverse/files/642979/output-54fa0ec.txt)

Here's where I made changes to the guides:

- http://guides.dataverse.org/en/3303-3515-installer-vagrant/installation/prep.html#vagrant-for-testing-only
- http://guides.dataverse.org/en/3303-3515-installer-vagrant/developers/tools.html#vagrant

**Comment by pdurbin** [comment]
Ok, I merged #3520 after consulting with @djbrooke and @kcondon . I'm convinced the installer works. Thanks @pameyer for pointing out the vagrant dependency on installing Shibboleth, which I removed. I also cleaned up the language in the guides. Closing.
